src/qibo/quantum_info/basis.py

```diff
@@ -1,12 +1,25 @@
+from functools import cache
 from itertools import product
 from typing import Optional
 
 import numpy as np
 
-from qibo import matrices
-from qibo.backends import _check_backend
+from qibo.backends import Backend, _check_backend
 from qibo.config import raise_error
-from qibo.quantum_info.superoperator_transformations import vectorization
+
+
+@cache
+def _get_paulis(order: str, backend: Backend):
+    pauli_labels = {"I": backend.matrices.I()}
+    pauli_labels.update(
+        {label: getattr(backend.matrices, label) for label in ("X", "Y", "Z")}
+    )
+    return [pauli_labels[label] for label in order]
+
+
+@cache
+def _normalization(nqubits: int):
+    return np.sqrt(2**nqubits)
```

Comment:
```suggestion
    return float(np.sqrt(2**nqubits))
```

Maybe it's best to force the `python` `float` type (or doing that by directly using `math.sqrt`) than using `numpy.float64`.